ros_industrial_cmake_boilerplate/cmake/cmake_tools.cmake

```diff
@@ -228,8 +263,9 @@ endmacro()
 #   * It create and install the ${PROJECT_NAME}-config.cmake and ${PROJECT_NAME}-config-version.cmake
 macro(generate_package_config)
   set(options EXPORT)
-  set(oneValueArgs NAMESPACE)
-  cmake_parse_arguments(ARG "${options}" "${oneValueArgs}" "" ${ARGN})
+  set(oneValueArgs NAMESPACE CONFIG_TEMPLATE)
+  set(multiValueArgs DEPENDENCIES)
+  cmake_parse_arguments(ARG "${options}" "${oneValueArgs}" "${multiValueArgs}" ${ARGN})
```

Comment:
Also it may be worth add another multiValueArgs `CFG_EXTRAS` similar to the ros which allow you to include extra configs to be included.

``` cmake
foreach(extra_config IN LISTS ARG_CFG_EXTRAS)
  list(APPEND include_extras_cmds "include(\"\${extra_config}\"\)")
endforeach()
```

Reply:
Actually this foreach loop would go in the function above.

Reply:
Yes, I think I can include this without too much effort.

Reply:
This is working now and lets the config file for ricb itself be autogenerated.